### ISSUE TITLE
make log_dict prog_bar accept loss keys to prevent the it from being too long.

### ISSUE THREAD

**Issue by tshu-w** [PyTorchLightning/pytorch-lightning#7843]
## 🚀 Feature
It will be convenient if LightningModule.log_dict pars prog_bar accept loss keys to prevent the prog_bar from being too long.

**Comment by awaelchli** [comment]
Hi
It is not clear to me what you mean by that. Can you please add more description to the feature request and perhaps give an example of what you wish to be displayed.

**Comment by tshu-w** [comment]
Sorry for the unclear description. I'm using `MetricCollection` and `self.log_dict`. However, too many metrics can make the prog bar incomplete. I was wondering if we could make prog_bar accept the key list of dict, like `self.log(metrics, on_epoch=True, prog_bar=["*_f1", "*_loss", "*_f1_step", "*_loss_step"])`. Of course, users can also extract the fields they want to display individually, but I think it's a bit less elegant. Or is there any other idea to solve the problem of incomplete display of `prog_bar`. 

<img width="768" alt="Screen Shot 2021-06-07 at 14 32 23" src="https://user-images.githubusercontent.com/13161779/120969827-3a7d1180-c79d-11eb-895e-d8601da9b0ff.png">

```python
def __init__():
    ...
    metrics = MetricCollection(
        {
            "f1": F1(**metrics_kwargs),
            "prc": Precision(**metrics_kwargs),
            "rec": Recall(**metrics_kwargs),
        }
    )
    ...

def training_step(self, batch, batch_idx):
    x, labels, _ = batch
    y = self.forward(x, labels)

    probs = F.softmax(y.logits, dim=-1)
    metrics = self.train_metrics(probs, labels)
    metrics["train_loss"] = y.loss

    self.log_dict(metrics, on_epoch=True, prog_bar=True)

    return y.loss
```

**Comment by tshu-w** [comment]
Oh, I find that we can modify the progress bar by `get_progress_bar_dict`.
